<h1>ACTIVITY 1 - PALER PENGUINS DATA ANALYTICS PROJECT</h1>

<hr>

<h3>ANALYTICS : KATRINA T. DAPATNAPO</h3>

In [ ]:
#import libraries
import numpy as np
import pandas as pd
from scipy.stats import f_oneway



<h2>Requirments</h2>

<ul>
    <li>Name of the DataFrame</li>
    <li>File Path</li>
</ul>

In [ ]:
penguins = pd.read_csv('Database\\penguins.csv')

In [ ]:
penguins

In [ ]:
penguins.isnull()

In [ ]:
penguins.isnull().sum()

In [ ]:
penguins['species']

In [ ]:
penguins['species'].value_counts()

In [8]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

In [9]:
penguins[penguins['species'] != 'Chinstrap']

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
271    272  Gentoo     Biscoe             NaN            NaN   
272    273  Gentoo     Biscoe            46.8           14.3   
273    274  Gentoo     Biscoe            50.4           15.7   
274    275  Gentoo     Biscoe            45.2           14.8   
275    276  Gentoo     Biscoe            49.9           16.1   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
271                NaN          NaN     NaN  2009  
272              215.0       4850.0  female  2009  
273              222.0       5750.0    male  2009  
274              212.0       5200.0  female  2009  
275              213.0       5400.0    male  2009  

[276 rows x 9 columns]

In [10]:
penguins[penguins['species'] == 'Chinstrap']

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [11]:
penguins[penguins['species'] == 'Adelie']

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
147    148  Adelie      Dream            36.6           18.4   
148    149  Adelie      Dream            36.0           17.8   
149    150  Adelie      Dream            37.8           18.1   
150    151  Adelie      Dream            36.0           17.1   
151    152  Adelie      Dream            41.5           18.5   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
147              184.0       3475.0  female  2009  
148              195.0       3450.0  female  2009  
149              193.0       3750.0    male  2009  
150              187.0       3700.0  female  2009  
151              201.0       4000.0    male  2009  

[152 rows x 9 columns]

In [12]:
penguins[penguins['species'] == 'Gentoo']

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152    153  Gentoo  Biscoe            46.1           13.2              211.0   
153    154  Gentoo  Biscoe            50.0           16.3              230.0   
154    155  Gentoo  Biscoe            48.7           14.1              210.0   
155    156  Gentoo  Biscoe            50.0           15.2              218.0   
156    157  Gentoo  Biscoe            47.6           14.5              215.0   
..     ...     ...     ...             ...            ...                ...   
271    272  Gentoo  Biscoe             NaN            NaN                NaN   
272    273  Gentoo  Biscoe            46.8           14.3              215.0   
273    274  Gentoo  Biscoe            50.4           15.7              222.0   
274    275  Gentoo  Biscoe            45.2           14.8              212.0   
275    276  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 9 columns]

In [13]:
penguins ['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [14]:
penguins[penguins['body_mass_g'] >= 4000]

rowid    species     island  bill_length_mm  bill_depth_mm  \
7        8     Adelie  Torgersen            39.2           19.6   
9       10     Adelie  Torgersen            42.0           20.2   
14      15     Adelie  Torgersen            34.6           21.1   
17      18     Adelie  Torgersen            42.5           20.7   
19      20     Adelie  Torgersen            46.0           21.5   
..     ...        ...        ...             ...            ...   
323    324  Chinstrap      Dream            49.0           19.6   
329    330  Chinstrap      Dream            50.7           19.7   
333    334  Chinstrap      Dream            49.3           19.9   
339    340  Chinstrap      Dream            55.8           19.8   
342    343  Chinstrap      Dream            50.8           19.0   

     flipper_length_mm  body_mass_g   sex  year  
7                195.0       4675.0  male  2007  
9                190.0       4250.0   NaN  2007  
14               198.0       4400.0  male  2007  
17               197.0       4500.0  male  2007  
19               194.0       4200.0  male  2007  
..                 ...          ...   ...   ...  
323              212.0       4300.0  male  2009  
329              203.0       4050.0  male  2009  
333              203.0       4050.0  male  2009  
339              207.0       4000.0  male  2009  
342              210.0       4100.0  male  2009  

[177 rows x 9 columns]

In [15]:
chinstrap_heavy = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

In [16]:
chinstrap_heavy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              16 non-null     int64  
 1   species            16 non-null     object 
 2   island             16 non-null     object 
 3   bill_length_mm     16 non-null     float64
 4   bill_depth_mm      16 non-null     float64
 5   flipper_length_mm  16 non-null     float64
 6   body_mass_g        16 non-null     float64
 7   sex                16 non-null     object 
 8   year               16 non-null     int64  
dtypes: float64(4), int64(2), object(3)
memory usage: 1.2+ KB


In [17]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [18]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean)

<h2>Changing the Data type</h2>

In [19]:
penguins['species'] = penguins['species'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [20]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    int64   
 1   species            344 non-null    category
 2   island             344 non-null    object  
 3   bill_length_mm     344 non-null    object  
 4   bill_depth_mm      342 non-null    float64 
 5   flipper_length_mm  342 non-null    float64 
 6   body_mass_g        342 non-null    float64 
 7   sex                333 non-null    object  
 8   year               344 non-null    category
dtypes: category(2), float64(3), int64(1), object(3)
memory usage: 19.9+ KB


<h2>Explenatory Data Analytics</h2>
<hr>

<h3>Question 1</h3>
<h4> Which specie has the longest flipper ?</h4>

In [21]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [22]:
gentoo_pdf = penguins[penguins['species'] == 'Gentoo']

In [23]:
gentoo_pdf['flipper_length_mm'].mean()

np.float64(217.1869918699187)

In [24]:
adelie_pdf = penguins[penguins['species'] == 'Adelie']

In [25]:
adelie_pdf['flipper_length_mm'].mean()

np.float64(189.95364238410596)

In [26]:
chinstrap_pdf = penguins[penguins['species'] == 'Chinstrap']

In [27]:
chinstrap_pdf['flipper_length_mm'].mean()

np.float64(195.8235294117647)

<h4>Insite No.1</h4>
<p>Gentoo has the longest Flipper lenght</p>

<h3>Question 2</h3>
<h4> What year has the most penguins ?</h4>

In [28]:
penguins['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h4>Insite No.2</h4>
<p>Year 2009 has the most penguins at that time </p>

<h3>Question 3</h3>
<h4>Which is more numerous, male or female? </h4>

In [29]:
penguins['sex'].value_counts()

sex
male      168
female    165
Name: count, dtype: int64

<h3>Insite 3</h3>
<p>According to the graph Male has the most number of sex</p>

<h3>Question 4</h3>
<h4>Which has most species</h4>

In [30]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h3>Insite 4</h3>
<p

<h3>Question 5</h3>
<h4>How many Males and Females are there in Gentoo specie?</h4>

In [31]:
f_counts = penguins[penguins['species'] == 'Gentoo']

In [32]:
f_counts['sex'].value_counts()

sex
male      61
female    58
Name: count, dtype: int64

<h3>Insite 5</h3>
<p>In Gentoo specie there are 61 males and 58 females</p>

<h3>Question 6</h3>
<h4>Which of the 3 has the most Male's ?</h4>

In [33]:
gm = penguins[penguins['species'] == 'Gentoo']

In [34]:
gm['sex'].value_counts()

sex
male      61
female    58
Name: count, dtype: int64

In [35]:
cm = penguins[penguins['species'] == 'Chinstrap']

In [36]:
cm['sex'].value_counts()

sex
female    34
male      34
Name: count, dtype: int64

In [37]:
am = penguins[penguins['species'] == 'Adelie']

In [38]:
am['sex'].value_counts()

sex
male      73
female    73
Name: count, dtype: int64

<h3>Insite 6</h3>
<p>As we can see Adelie has the most male among the 3 species</p>

<h3>Question 7</h3>
<h4>What are the common bill length on a penguins ?</h4>

In [39]:
penguins['bill_length_mm'].mode()

0    41.1
Name: bill_length_mm, dtype: object

<h3>Insite 7</h3>
<p>According to the calculation most of the penguins have 41.1 bill length</p>

<h3>Question 8</h3>
<h4>Does the body mass affect the penguins flipper length?</h4>

In [40]:
fs = penguins ['body_mass_g'].corr(penguins['flipper_length_mm'])

In [41]:
fs

np.float64(0.8712017673060113)

In [42]:
from scipy.stats import f_oneway

<h3>Insite 8 </h3>
<p>The body mass does affect the flipper length</p>

<h3>Question 9</h3>
<h4>Does the gender of the penguins affects their body mass?</h4>

In [49]:
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.0
5      3650.0
7      4675.0
13     3800.0
14     4400.0
        ...  
334    3800.0
336    3950.0
339    4000.0
341    3775.0
342    4100.0
Name: body_mass_g, Length: 168, dtype: float64

In [59]:
no_use, p_value = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

In [60]:
p_value

np.float64(4.897246751596225e-16)

<h3>Insite 9</h3>
<p>The sex of the penguins does affect their body mass</p>

<h3>Question 10</h3>
<h4>Does the body mass affect the bill length of the penguin?</h4>

In [61]:
penguins[penguins['species'] == 'Chinstrap']['bill_length_mm']

276    46.5
277    50.0
278    51.3
279    45.4
280    52.7
       ... 
339    55.8
340    43.5
341    49.6
342    50.8
343    50.2
Name: bill_length_mm, Length: 68, dtype: object

In [64]:
no_use, f_value = f_oneway(penguins[penguins['species'] == 'Chinstrap']['bill_length_mm'],penguins[penguins['species'] == 'Chinstrap']['body_mass_g'])

In [65]:
f_value

np.float64(2.66131375510223e-114)

<h3>Insite 10</h3>
<p>YES, The body mass affects the bill length of the penguin</p>

<h3>Question 11</h3>
<h4>How many Females are ther in Gentoo species in year 2008?</h4>

In [73]:
per_island = penguins.groupby(['species','year','sex']).size()

C:\Program Files\KMSpico\temp\ipykernel_5540\3091007729.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  per_island = penguins.groupby(['species','year','sex']).size()


In [74]:
per_island 

species    year  sex   
Adelie     2007  female    22
                 male      22
           2008  female    25
                 male      25
           2009  female    26
                 male      26
Chinstrap  2007  female    13
                 male      13
           2008  female     9
                 male       9
           2009  female    12
                 male      12
Gentoo     2007  female    16
                 male      17
           2008  female    22
                 male      23
           2009  female    20
                 male      21
dtype: int64

<h3>Insite 11</h3>
<p>At year 2008 ther are 22 female Gentoo species </p>

<h3>Question 12</h3>
<h4>Does the sex of the penguin specie affects their bill length?</h4>


In [80]:
penguins[penguins['sex'] == 'male']['bill_length_mm']

0      39.1
5      39.3
7      39.2
13     38.6
14     34.6
       ... 
334    50.2
336    51.9
339    55.8
341    49.6
342    50.8
Name: bill_length_mm, Length: 168, dtype: object

In [86]:
no_use, b_value = f_oneway(penguins[penguins['sex'] == 'male']['bill_length_mm'],penguins[penguins['sex'] == 'female']['bill_length_mm'])

In [87]:
b_value

np.float64(1.0942555387199773e-10)

<h3>Insite 12</h3>
<p>YES, the sex of the penguins affects their bill length</p>

<h3>Question 13</h3>
<h4>How many Male Adelie are there at year 2009?</h4>

In [91]:
spe_island = penguins.groupby(['species','year','sex']).size()

C:\Program Files\KMSpico\temp\ipykernel_5540\1914589255.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  spe_island = penguins.groupby(['species','year','sex']).size()


In [92]:
spe_island

species    year  sex   
Adelie     2007  female    22
                 male      22
           2008  female    25
                 male      25
           2009  female    26
                 male      26
Chinstrap  2007  female    13
                 male      13
           2008  female     9
                 male       9
           2009  female    12
                 male      12
Gentoo     2007  female    16
                 male      17
           2008  female    22
                 male      23
           2009  female    20
                 male      21
dtype: int64

<h3>Insite 13</h3>
<p>There are 26 male chinstrap species at year 2009</p>